code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -43,31 +43,30 @@
 	if(I.resistance_flags & INDESTRUCTIBLE)
 		to_chat(user, "<span class='warning'>You don't feel it would be wise to grill [I]...</span>")
 		return ..()
-	if(istype(I, /obj/item/reagent_containers))
-		if(istype(I, /obj/item/reagent_containers/food) && !istype(I, /obj/item/reagent_containers/food/drinks))
-			var/obj/item/reagent_containers/food/food_item = I
-			if(HAS_TRAIT(food_item, TRAIT_NODROP) || (food_item.item_flags & (ABSTRACT | DROPDEL)))
-				return ..()
-			else if(food_item.foodtype & GRILLED)
-				to_chat(user, "<span class='notice'>[food_item] has already been grilled!</span>")
-				return
-			else if(grill_fuel <= 0)
-				to_chat(user, "<span class='warning'>There is not enough fuel!</span>")
-				return
-			else if(!grilled_item && user.transferItemToLoc(food_item, src))
-				grilled_item = food_item
-				grilled_item.foodtype |= GRILLED
-				to_chat(user, "<span class='notice'>You put the [grilled_item] on [src].</span>")
-				update_icon()
-				grill_loop.start()
-				return
-		else
-			if(I.reagents.has_reagent(/datum/reagent/consumable/monkey_energy))
-				grill_fuel += (20 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy)))
-				to_chat(user, "<span class='notice'>You pour the Monkey Energy in [src].</span>")
-				I.reagents.remove_reagent(/datum/reagent/consumable/monkey_energy, I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy))
-				update_icon()
-				return
+	if(istype(I, /obj/item/reagent_containers/food/drinks))
+		if(I.reagents.has_reagent(/datum/reagent/consumable/monkey_energy))
+			grill_fuel += (20 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy)))
+			to_chat(user, "<span class='notice'>You pour the Monkey Energy in [src].</span>")
+			I.reagents.remove_reagent(/datum/reagent/consumable/monkey_energy, I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy))
+			update_icon()
+			return
```

Comment:
this whole thing is my dumb code with requiring specifically food/drinks and not allowing beakers
maybe just do if is edible and then else if reagent containers with this piece of code

Reply:
it was my dumb code so it would be oversight fix not balance change 😏

Reply:
understandable